core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -86,6 +72,84 @@
         public static final String FETCH_SHARD_STORE = "fetch_shard_store";
     }
 
+    public enum ThreadPoolType {
```

Comment:
Nevermind, we already have docs for it in the actual docs, so we can leave it out here.